hazelcast/src/test/java/com/hazelcast/jet/pipeline/SourcesTest.java

```diff
@@ -244,6 +252,148 @@ public void remoteMap() {
+
+    @Test
+    public void remoteReplicatedMap_dataConnectionName_customBatchSize() {
+        remoteReplicatedMap_dataConnectionName((dataConnectionName) -> Sources.remoteReplicatedMap(srcName, dataConnectionName, 100));
+    }
+
+    @Test
+    public void remoteReplicatedMap_emptySourceMap() {
+        remoteReplicatedMap_emptySourceMap((dataConnectionName) -> Sources.remoteReplicatedMap(srcName, dataConnectionName));
+    }
+
+    @Test
+    public void remoteReplicatedMap_emptySourceMap_customBatchSize() {
+        remoteReplicatedMap_emptySourceMap((dataConnectionName) -> Sources.remoteReplicatedMap(srcName, dataConnectionName, 100));
+    }
+
+    private void remoteReplicatedMap_emptySourceMap(Function<String, BatchSource<Entry<Object, Object>>> sourceFn) {
+        // Given empty map
+        // When
+        String dataConnectionName = randomString();
+        hz().getConfig().addDataConnectionConfig(new DataConnectionConfig(dataConnectionName)
+                .setType("Hz")
+                .setShared(false)
+                .setProperty(HazelcastDataConnection.CLIENT_XML, ImdgUtil.asXmlString(clientConfig)));
+        BatchSource<Entry<Object, Object>> source = sourceFn.apply(dataConnectionName);
+
+        // Then
+        p.readFrom(source).writeTo(sink);
+        execute();
+        assertThat(sinkList).isEmpty();
+    }
+
+    @Test
+    public void remoteReplicatedMap_dataConnectionMissing() {
+        remoteReplicatedMap_dataConnectionMissing((dataConnectionName) -> Sources.remoteReplicatedMap(srcName, dataConnectionName));
+    }
+
+    @Test
+    public void remoteReplicatedMap_dataConnectionMissing_customBatchSize() {
+        remoteReplicatedMap_dataConnectionMissing((dataConnectionName) -> Sources.remoteReplicatedMap(srcName, dataConnectionName, 100));
+    }
+
+    private void remoteReplicatedMap_dataConnectionMissing(Function<String, BatchSource<Entry<Object, Object>>> sourceFn) {
+        // Given
+        List<Integer> input = sequence(itemCount);
+        putToMap(remoteHz.getReplicatedMap(srcName), input);
+
+        // When
+        String dataConnectionName = randomString();
+        BatchSource<Entry<Object, Object>> source = sourceFn.apply(dataConnectionName);
+
+        // Then
+        p.readFrom(source).writeTo(sink);
+        assertThatThrownBy(this::execute).isInstanceOf(CompletionException.class)
+                .hasStackTraceContaining("Data connection '" + dataConnectionName + "' not found");
+    }
+
+    @Test
+    public void remoteReplicatedMap_dataConnectionToNonExistentCluster() {
+        remoteReplicatedMap_dataConnectionToNonExistentCluster((dataConnectionName) -> Sources.remoteReplicatedMap(srcName, dataConnectionName));
+    }
+
+    @Test
+    public void remoteReplicatedMap_dataConnectionToNonExistentCluster_customBatchSize() {
+        remoteReplicatedMap_dataConnectionToNonExistentCluster((dataConnectionName) -> Sources.remoteReplicatedMap(srcName, dataConnectionName, 100));
+    }
```

Comment:
Similarly - does this test something specifically?

Reply:
right, maybe it makes sense to remove batch size versions of the test for other tests than the simple test case where we test if it works fine.